src/LuaTransformer.ts

```diff
@@ -1229,7 +1238,7 @@ export class LuaTransformer {
                 restParamName,
                 tstl.FunctionExpressionFlags.Declaration
             ),
-            statement
+            statement.pos >= 0 ? statement : classDeclaration
```

Comment:
This seems strange, either move this to a variable with descriptive name or add a comment with explanation.